include/like.php

```diff
@@ -73,28 +73,33 @@ function do_like($item_id, $verb) {
 		dbesc($item_id)
 	);
 
-	if (! $item_id || ! DBM::is_result($items)) {
+	if (!$item_id || !DBM::is_result($items)) {
 		logger('like: unknown item ' . $item_id);
 		return false;
 	}
 
 	$item = $items[0];
+	$uid = $item['uid'];
 
-	if (! can_write_wall($a, $item['uid'])) {
-		logger('like: unable to write on wall ' . $item['uid']);
+	if (($uid == 0) && local_user()) {
+		$uid = local_user();
```

Comment:
I can’t see the rest of the code on mobile so I’m a little puzzled: the public item is owned by `uid = 0` and the likes seem to be owned by `local_user()`. How is it going to work for display, don’t we retrieve just the items from a single `uid`?

Reply:
That's why the item is posted with the "uid=0", but the contact-id with "uid=local_user()".

Reply:
Ok good, thanks for the clarification.